application.go

```diff
@@ -272,11 +351,45 @@ func (r *Application) CheckTCP(port, interval int) (*Application, error) {
 	health.IntervalSeconds = interval
 	health.PortIndex = portIndex
 	// step: add to the checks
-	r.HealthChecks = append(r.HealthChecks, health)
+	r.AddHealthCheck(*health)
 
 	return r, nil
 }
 
+// AddUris adds one or more uris to the applications
+//		arguments:	the uri(s) you are adding
+func (r *Application) AddUris(newUris ...string) *Application {
+	if r.Uris == nil {
+		r.EmptyUris()
+	}
+
+	uris := *r.Uris
+	uris = append(uris, newUris...)
+	r.Uris = &uris
+
+	return r
+}
+
+// EmptyUris explicitly empties uris -- use this if you need to empty
+// uris of an application that already has uris set (setting uris to nil will
+// keep the current value)
+func (r *Application) EmptyUris() *Application {
+	empty := []string{}
+	r.Uris = &empty
+
+	return r
+}
+
+// String returns the json representation of this application
+func (r *Application) String() string {
```

Comment:
Where do we use this?

If we do somewhere: I believe `String()` should be reserved for stringifying a Go type, so this one may be better called `JSON()`. It's also missing test coverage.